app/javascript/components/widgets/weather/weather-effect.jsx

```diff
@@ -345,61 +346,62 @@ const DrizzleLight = () => <Drizzle duration={1.8} variance={0.5} count={30} />;
-  621: <Snow duration={4} variance={1} count={50} />, // Normal
-  622: <Snow duration={2} variance={1} count={70} />, // Heavy
-  801: <Cloud duration={16} variance={1} count={2} />, // few cloud
-  802: <Cloud duration={16} variance={5} count={4} />, // scattered cloud
-  803: <Cloud duration={16} variance={10} count={8} />, // broken cloud
-  804: <Cloud duration={16} variance={15} count={16} />, // overcast cloud
-}
-/* eslint-enable prettier/prettier */
 
 const WeatherEffect = ({ weather }) => {
   const {
     current: {
       weather: { id },
+      windSpeed
     },
   } = weather;
 
+  const weatherIdMap = {
+    200: <><RainModerate /> <Thunder /></>, // Light rain
+    201: <><RainHeavy /> <Thunder /></>, // Moderate rain
+    202: <><RainIntense /> <Thunder /></>, // Heavy rain
+    210: <Thunder />, // Light thunderstorm
+    211: <Thunder />, // Moderate thunderstorm
+    212: <Thunder />, // Heavy thunderstorm
+    221: <Thunder />, // Ragged thunderstorm
+    230: <><DrizzleLight /> <Thunder /></>, // Light drizzle
+    231: <><DrizzleModerate /> <Thunder /></>, // Moderate drizzle
+    232: <><DrizzleHeavy /> <Thunder /></>, // Heavy drizzle
+    300: <DrizzleLight />, // Light
+    301: <DrizzleModerate />, // Moderate
+    302: <DrizzleHeavy />, // Heavy
+    310: <><DrizzleLight /> <RainLight /></>, // Light rain
+    311: <><DrizzleLight /> <RainModerate /></>, // Moderate rain
+    312: <><DrizzleModerate /> <RainModerate /></>, // Heavy rain
+    313: <><DrizzleLight /> <RainModerate /></>, // Shower rain
+    314: <><DrizzleModerate /> <RainModerate /></>, // Heavy shower rain
+    321: <DrizzleModerate />, // Shower
+    500: <RainLight />, // Light
+    501: <RainModerate />, // Moderate
+    502: <RainHeavy />, // Heavy
+    503: <RainIntense />, // Very heavy
+    504: <RainExtreme />, // Extreme
+    511: <RainModerate />, // Freezing
+    520: <RainLight />, // Light shower
+    521: <RainModerate />, // Moderate shower
+    522: <RainHeavy />, // Heavy shower
+    531: <RainModerate />, // Ragged shower
+    600: <Snow duration={6} variance={2} count={20} />, // Light
+    601: <Snow duration={4} variance={2} count={50} />, // Normal
+    602: <Snow duration={2} variance={2} count={70} />, // Heavy
+    611: <HeavySnow duration={3} variance={2} count={20} />, // Sleet
+    612: <HeavySnow duration={2} variance={1} count={35} />, // Sleet
+    613: <HeavySnow duration={1.5} variance={1} count={45} />, // Sleet
+    615: <><Snow duration={4} variance={2} count={20} /> <RainLight /></>, // Light rain and snow
+    616: <><Snow duration={2} variance={2} count={30} /> <RainModerate /></>, // Rain and snow
+    620: <Snow duration={6} variance={1} count={20} />, // Light
+    621: <Snow duration={4} variance={1} count={50} />, // Normal
+    622: <Snow duration={2} variance={1} count={70} />, // Heavy
+    801: <Cloud duration={windSpeed * 100} variance={1} count={2} />, // few cloud
```

Comment:
So it looks like you ran into a problem here trying to inject a property acquired from the graphql query into the constant component renders in `weatherIdMap`. This is probably the simplest solution by presents some problems:
* We've moved a lot of code that was not related to our feature
* `weatherIdMap` is rebuilt on every render removing the benefit of `weatherIdMap`
* The `WeatherEffect` render is very long now

Some other approaches:
* Leave `weatherIdMap` defined above and split out a condition for `801`, `802`, `803`, `804`. This is slightly simpler approach but breaks our original pattern. It is very likely we will want windSpeed (maybe other values) in our other weather effects so lets skip this.
* React provides an API `cloneElement` that can take an existing element and extend it with additional props: https://reactjs.org/docs/react-api.html#cloneelement This keeps our original pattern but likely does not do what we want for the weather effects in fragments: <><RainModerate /> <Thunder /></>
* Rethink our original pattern to suit our new desired functionality. We could switch all existing entries (refactor commit) to a new pattern of
```
{
  200: [{ type: RainModerate: props: {}}, { type: Thunder, props: {}}],
  ...
  600: [{ type: Snow, props: { duration: 6, variance: 2, count: 20 }} ],
}
```
and later our in `WeatherEffect`:
```
const Effects = weatherIdMap[id];
return Effects ? <EffectContainer>{Effects.map(({ type, props }) => <type {...props} />)}</EffectContainer> : null;
```

After the refactor, you're now free to inject `windSpeed` (and any other piece of weather data) into the rendered effects.